app/controllers/workflows_controller.rb

```diff
@@ -86,9 +86,9 @@ def update
 
       workflow.update(workflow_params)
 
-      if (was_paused || was_halted) && workflow.active?
-        UnpauseWorkflowWorker.perform_async workflow.id
-        flash[:notice] = 'Resuming workflow'
+      if workflow.active?
+        flash[:notice] = 'Resuming workflow' if was_paused || was_halted
+        UnpauseWorkflowWorker.perform_async(workflow.id) if was_paused && !was_halted
```

Comment:
Since status is an enum, it isn't possible for a workflow to be paused && halted. I think `if was_paused` is sufficient here.